src/Symfony/Component/Config/Tests/Resource/ReflectionClassResourceTest.php

```diff
@@ -136,6 +136,26 @@ public function provideHashedSignature()
         yield [0, 15, ''];
     }
 
+    /**
+     * Typed properties without a default value are undefined,
+     * make sure this doesn't throw an error when generating a signature.
+     *
+     * @depends testHashedSignature
+     * @dataProvider provideTypedProperties
+     * @requires PHP >= 7.4
+     */
+    public function testTypedProperies($changeExpected, $changedLine, $changedCode)
+    {
+        $this->testHashedSignature($changeExpected, $changedLine, $changedCode);
+    }
+
+    public function provideTypedProperties()
+    {
+        yield [1, 5, 'public array $pub;'];
```

Comment:
What if we updated the `provideHashedSignature()` method instead and yield these three examples only if `PHP_VERSION_ID >= 70400` is `true`?

Reply:
that works as well, and is a little cleaner indeed, I've updated the test.